types.go

```diff
@@ -250,17 +222,6 @@ func isDisk(t types.Type) bool {
 	return false
 }
 
-func isTimerRef(t types.Type) bool {
-	if t, ok := t.(*types.Pointer); ok {
-		if t, ok := t.Elem().(*types.Named); ok {
-			name := t.Obj()
-			return name.Pkg().Path() == "github.com/mit-pdos/gokv/time" &&
-				name.Name() == "Timer"
-		}
-	}
-	return false
-}
```

Comment:
I am not quite sure why, but this function is unused now -- similar functions like `isLockRef` are used in `goose.go` itself, but this one is not.

Reply:
Above in `ptrType` there were a bunch of special cases and this was one of them. Is it correct to replace all of them (`lockRefT` and `condvarRefT` and such) with `coq.PtrType`? They probably mean the same thing now but the names do create a little bit of an abstraction barrier.

Reply:
I checked all of them and they are all pointers, so it seemed unnecessary to have Goose distinguish them. I originally thought this could entirely get rid of `isLockRef` et al, but that did not work out.

I can reinstate those types if you prefer (and just change their Coq-level definitions to `ptrT`).

Reply:
Let's keep the PR as it is, I don't think it adds much to distinguish between the special references and other pointers. If we supported `sync.Mutex` directly inside structs that would be another story.